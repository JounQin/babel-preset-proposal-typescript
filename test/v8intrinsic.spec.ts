import { execSync } from 'child_process'
import fs from 'fs'

import tempy from 'tempy'

import { execute, read, resolve } from './helpers'

import { transform } from '@babel/core'

const proposal = 'v8intrinsic'

test(proposal, () => {
  expect(() => execute(proposal)).toThrowErrorMatchingSnapshot()
  const transformed = transform(read(proposal), {
    filename: resolve(proposal),
  }).code
  expect(transformed).toContain('%GetOptimizationStatus(fn)')
  const tempFile = tempy.file({ extension: 'js' })
  fs.writeFileSync(tempFile, transformed)
  const output = execSync('node --allow-natives-syntax ' + tempFile).toString()
  fs.unlinkSync(tempFile)
  expect(output).toBe('1')
})
