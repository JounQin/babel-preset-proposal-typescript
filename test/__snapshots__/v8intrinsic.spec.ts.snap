// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v8intrinsic 1`] = `"Unexpected token %"`;
